sparksteps/cluster.py

```diff
@@ -54,6 +55,21 @@ def parse_conf(raw_conf_list):
     return defaults
 
 
+def parse_apps(raw_app_list):
+    """
+    Given a list of app name strings,
+    returns formatted application configuration value.
+
+    Examples:
+        >>> from pprint import pprint
+        >>> pprint(parse_apps(['hadoop', 'spark']))
+        [{'Name': 'Hadoop', 'Name': 'Spark'}]
+    """
+    return sorted(
+        [{'Name': app_name.capitalize()} for app_name in set(raw_app_list)],
```

Comment:
Interesting idea. I wonder if there's too much magic here where we autocapitalize.

Reply:
It should work as expected in all cases:

```python
'spark'.capitalize() == 'Spark'
'SPARK'.capitalize() == 'Spark'
'Spark'.capitalize() == 'Spark'
```